tests/arguments_test.py

```diff
@@ -0,0 +1,395 @@
+    # test_files/valid_config.yml
+    valid_yml_translated = {
+        "site_energy": True,
+        "captured_energy": False,
+        "alt_regions": "EMM",
+        "tsv_type": "energy",
+        "tsv_daily_hr_restrict": "all",
+        "tsv_sys_shape_case": "total reference",
+        "tsv_season": "summer",
+        "tsv_energy_agg": "average",
+        "tsv_power_agg": None,
+        "tsv_average_days": "weekdays",
+        "sect_shapes": False,
+        "rp_persist": True,
+        "verbose": False,
+        "health_costs": False,
+        "split_fuel": False,
+        "no_scnd_lgt": False,
+        "floor_start": None,
+        "pkg_env_costs": "1",
+        "exog_hp_rate_scenario": "aggressive",
+        "switch_all_retrofit_hp": False,
+        "alt_ref_carb": False,
+        "grid_decarb_level": "full",
+        "grid_assesment_timing": "after",
+        "adopt_scn_restrict": ["Max adoption potential"],
+        "retrofit_type": "increasing",
+        "retrofit_multiplier": 1.2,
+        "retrofit_mult_year": 2030,
+        "add_typ_eff": True,
+        "pkg_env_sep": False,
+        "detail_brkout": "6",
+        "fugitive_emissions": [3, 1],
+        "retro_set": ["3", 1.2, 2030],
+        "exog_hp_rates": ["aggressive", "2"],
+        "grid_decarb": ["1", "2"],
+        "tsv_metrics": ["1", "1", "1", "2", "1", "2"],
+    }
+
+    def test_translate_empty_cli(self):
+        # Translation of empty/default arguments
+        args = ecm_args([])
+        self._assert_arg_vals(args, self.default_translated)
+
+    def test_translate_valid_cli(self):
+        # Translation of valid cli inputs
+        cli_args = [
+            "--add_typ_eff",
+            "--alt_regions",
+            "EMM",
+            "--detail_brkout",
+            "regions",
+            "fuel types",
+        ]
+        args = ecm_args(cli_args)
+        self.assertEqual(args.add_typ_eff, True)
+        self.assertEqual(args.alt_regions, "EMM")
+        self.assertEqual(args.detail_brkout, "6")
+
+        cli_args = [
+            "--detail_brkout",
+            "buildings",
+            "--retrofit_type",
+            "increasing",
+            "--retrofit_multiplier",
+            "2",
+            "--retrofit_mult_year",
+            "2030",
+        ]
+        args = ecm_args(cli_args)
+        self.assertEqual(args.detail_brkout, "3")
+        self.assertEqual(args.retro_set, ["3", 2.0, 2030])
+
+    def test_translate_from_empty_cfg(self):
+        # Translation of empty yml (default args)
+        args = ecm_args(["--yaml", self.empty_yml_pth])
+        self._assert_arg_vals(args, self.default_translated)
+
+    def test_translate_from_valid_cfg(self):
+        # Translation of valid yaml args
+        args = ecm_args(["--yaml", self.valid_yml_pth])
+        self._assert_arg_vals(args, self.valid_yml_translated)
+
+    def test_translate_cli_overwrite(self):
+        # Ensure cli arguments take precedence over yml
```

Comment:
```suggestion
        # Ensure cli arguments take precedence over yml and are correctly translated
```

You could use different words here, but my initial reaction was "how is this test different from `TestConfig.test_cli_overwrite()`" so the change is proposed to hopefully prevent that misinterpretation.